thoth-app/src/component/delete_dialogue.rs

```diff
@@ -0,0 +1,98 @@
+use crate::string::DELETE_BUTTON;
+use yew::html;
+use yew::prelude::*;
+
+pub struct ConfirmDeleteComponent {
+    props: Props,
+    link: ComponentLink<Self>,
+}
+
+#[derive(Clone, Properties)]
+pub struct Props {
+    pub onclick: Callback<MouseEvent>,
+    pub object_name: String,
+    #[prop_or(false)]
+    pub show: bool,
+}
+
+pub enum Msg {
+    ToggleConfirmDeleteDisplay(bool),
+}
+
+impl Component for ConfirmDeleteComponent {
+    type Message = Msg;
+    type Properties = Props;
+
+    fn create(props: Self::Properties, link: ComponentLink<Self>) -> Self {
+        ConfirmDeleteComponent { props, link }
+    }
+
+    fn change(&mut self, _props: Self::Properties) -> ShouldRender {
+        false
+    }
+
+    fn update(&mut self, msg: Self::Message) -> ShouldRender {
+        match msg {
+            Msg::ToggleConfirmDeleteDisplay(value) => {
+                self.props.show = value;
+                true
+            }
+        }
+    }
+
+    fn view(&self) -> Html {
+        let open_modal = self.link.callback(|e: MouseEvent| {
+            e.prevent_default();
+            Msg::ToggleConfirmDeleteDisplay(true)
+        });
+        let close_modal = self.link.callback(|e: MouseEvent| {
+            e.prevent_default();
+            Msg::ToggleConfirmDeleteDisplay(false)
+        });
+        html! {
+            <>
+                <button class="button is-danger" onclick=open_modal>
+                    { DELETE_BUTTON }
+                </button>
+                <div class=self.confirm_delete_status()>
+                    <div class="modal-background"></div>
```

Comment:
Could you add an option to dismiss the modal, e.g.

```
<div class="modal-background" onclick=&close_modal></div>
```

Reply:
I deliberately decided not to do this so that the user would have to actively choose either `Delete` or `Cancel` to leave the dialogue, rather than clicking somewhere and the modal closing and them not being sure which action had been taken. But I'm aware it makes it inconsistent with our other modal dialogues, and in practice it ought to be fairly clear that no deletion has taken place, so I'm happy to change this.

Reply:
I thought that might have been the case, but I normally find it annoying when certain sites do not implement standard behaviour, and it's already bad enough that we don't allow closing modals by pressing esc (we should add this functionality at some point)...

Reply:
The other dialogues also have a Close (`x`) button in the top right, so I added that back in too.